src/db/collections/outputs/mod.rs

```diff
@@ -468,6 +468,127 @@ impl MongoDb {
     }
 }
 
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct OutputDiffAnalyticsResult {
+    pub created_count: u64,
+    pub transferred_count: u64,
+    pub burned_count: u64,
+}
+
+impl MongoDb {
+    /// Gathers nft output analytics.
+    pub async fn get_nft_output_analytics(
+        &self,
+        start_index: Option<MilestoneIndex>,
+        end_index: Option<MilestoneIndex>,
+    ) -> Result<OutputDiffAnalyticsResult, Error> {
+        Ok(self
+            .db
+            .collection::<OutputDiffAnalyticsResult>(OutputDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "output.kind": "nft",
+                        "metadata.booked.milestone_index": { "$not": { "$gt": start_index } },
```

Comment:
Booked is always there, so can't we use `$lte`?

Reply:
This is not because booked can be null, but because `start_index` can.